client/helpers/field-change-detection.js

```diff
@@ -12,6 +12,10 @@ import { normaliseDuration } from './normalise-duration';
  * @returns {boolean} - `true` if a change is detected, otherwise `false`.
  */
 export function hasDatabaseChange(fieldName, storedValue, currentValue, values, hasSpeciesFieldChanges) {
+  // Prevents crashes if fieldName is undefined or invalid
+  if (!fieldName || typeof fieldName !== 'string') {
+    return false;
+  }
```

Comment:
For arrays this could be a number, we can probably rely on javascript converting this to a scalar automatically rather than type checking, also it would be a code error if the fieldName was missing as that comes from the schema.
```suggestion
  if fieldName == null {
    throw new Error('fieldName must be provided.');
  }
```